File: logic/card.js
let SUITS = ["clubs", "diamonds", "hearts", "spades"];

let VALUES = ["2","3","4","5","6","7","8","9","10","jack","queen","king","ace"];


class Card {
    constructor(suit, rank) {
        this.suit = suit;
        this.rank = rank;
        this.name = this.rank + " of " + this.suit;
        this.filename = this.rank + "_of_" + this.suit + ".png";
    }
    getValue() {
        switch(this.rank) {
            case "2":
                return 2;
            case "3":
                return 3;
            case "4":
                return 4;
            case "5":
                return 5;
            case "6":
                return 6;
            case "7":
                return 7;
            case "8":
                return 8;
            case "9":
                return 9;
            case "10":
                return 10;
            case "jack":
                return 11;
            case "queen":
                return 12;
            case "king":
                return 13;
            case "ace":
                return 14;  
        }
    }
    getSuit() {
        return this.suit;
    }
}


module.exports = Card;

File: python/card.py
# Colors
red = "\033[0;31m"
green = "\033[0;32m"
yellow = "\033[0;33m"
blue = "\033[0;34m"
purple = "\033[0;35m"
cyan = "\033[0;36m"
white = "\033[0;37m"
reset = "\033[0m"

class Card:
    def __init__(self, s, r):
        self.suit = s
        self.rank = r

        self.suits = {1: blue + "Spades" + reset, 2: red + "Diamonds" + reset, 3: blue + "Clubs" + reset, 4: red + "Hearts" + reset}
        self.ranks = {2: "Two", 3: "Three", 4: "Four", 5: "Five", 6: "Six", 7: "Seven", 8: "Eight", 9: "Nine", 10: "Ten", 11: green + "Jack" + reset, 12: purple + "Queen" + reset, 13: cyan + "King" + reset, 14: yellow + "Ace" + reset}

    def __str__(self):
        return f"{self.ranks[self.rank]} of {self.suits[self.suit]}"

    def showCard(self):
        print(str(self.ranks[self.rank]) + " of " + str(self.suits[self.suit]))


